RecoTracker/LSTCore/src/alpaka/NeuralNetwork.h

```diff
@@ -24,10 +25,11 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst::t5dnn {
   }
 
   template <int IN_FEATURES, int OUT_FEATURES>
-  ALPAKA_FN_ACC ALPAKA_FN_INLINE void linear_layer(const float (&input)[IN_FEATURES],
-                                                   float (&output)[OUT_FEATURES],
-                                                   const float (&weights)[IN_FEATURES][OUT_FEATURES],
-                                                   const float (&biases)[OUT_FEATURES]) {
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE void linear_layer(
+      const float (&input)[IN_FEATURES],
+      float (&output)[OUT_FEATURES],
+      const std::array<std::array<float, OUT_FEATURES>, IN_FEATURES>& weights,
```

Comment:
Line 31 here, changing the weight arrays to these 2d std array's, is what causes most of the timing increase on GPU.

Without changing this line.
![Screenshot 2025-01-07 at 5 55 33 PM](https://github.com/user-attachments/assets/8dd021d9-37e4-4d0e-a6f4-dea879111454)

After changing this line.
![Screenshot 2025-01-07 at 4 51 14 PM](https://github.com/user-attachments/assets/f03bc53b-fc56-4849-b8c2-50c00c1d9fe4)